src/java.desktop/share/classes/javax/swing/text/html/CSS.java

```diff
@@ -1391,58 +1394,179 @@ static final Color hexToColor(String value) {
      * or "rgba(r, g, b, a)" to a Color.
      */
     static Color stringToColor(String str) {
-      Color color;
-
-      if (str == null) {
-          return null;
-      }
-      if (str.length() == 0)
-        color = Color.black;
-      else if (str.startsWith("rgb(")) {
-          color = parseRGB(str);
-      } else if (str.startsWith("rgba(")) {
-          color = parseRGBA(str);
-      }
-      else if (str.charAt(0) == '#')
-        color = hexToColor(str);
-      else if (str.equalsIgnoreCase("Black"))
-        color = hexToColor("#000000");
-      else if(str.equalsIgnoreCase("Silver"))
-        color = hexToColor("#C0C0C0");
-      else if(str.equalsIgnoreCase("Gray"))
-        color = hexToColor("#808080");
-      else if(str.equalsIgnoreCase("White"))
-        color = hexToColor("#FFFFFF");
-      else if(str.equalsIgnoreCase("Maroon"))
-        color = hexToColor("#800000");
-      else if(str.equalsIgnoreCase("Red"))
-        color = hexToColor("#FF0000");
-      else if(str.equalsIgnoreCase("Purple"))
-        color = hexToColor("#800080");
-      else if(str.equalsIgnoreCase("Fuchsia"))
-        color = hexToColor("#FF00FF");
-      else if(str.equalsIgnoreCase("Green"))
-        color = hexToColor("#008000");
-      else if(str.equalsIgnoreCase("Lime"))
-        color = hexToColor("#00FF00");
-      else if(str.equalsIgnoreCase("Olive"))
-        color = hexToColor("#808000");
-      else if(str.equalsIgnoreCase("Yellow"))
-        color = hexToColor("#FFFF00");
-      else if(str.equalsIgnoreCase("Navy"))
-        color = hexToColor("#000080");
-      else if(str.equalsIgnoreCase("Blue"))
-        color = hexToColor("#0000FF");
-      else if(str.equalsIgnoreCase("Teal"))
-        color = hexToColor("#008080");
-      else if(str.equalsIgnoreCase("Aqua"))
-        color = hexToColor("#00FFFF");
-      else if(str.equalsIgnoreCase("Orange"))
-        color = hexToColor("#FF8000");
-      else
-          color = hexToColor(str); // sometimes get specified without leading #
-      return color;
-    }
+        String strlc = str.toLowerCase(Locale.ROOT);
+        if (str == null) {
+            return null;
+        } else if (str.isEmpty()) {
+            return Color.black;
+        } else if (strlc.startsWith("rgb(")) {
+            return parseRGB(str);
+        } else if (strlc.startsWith("rgba(")) {
+            return parseRGBA(str);
+        } else if (str.charAt(0) == '#') {
+            return hexToColor(str);
+        } else {
+            Color color = colorNamed.get(strlc);
+            if (color != null) {
+                return new Color(color.getRGB(), true);
```

Comment:
Can you explain why you can't return the color instance directly ?
They are immutable so I don't see the reason.

Reply:
To avoid change method behaviour, I tried to reproduce previous behaviors because returned objects can be publicly exposed trough javax.swing.text.html.StyleSheet stringToColor method.
Previous implementation :
When, "" returns always the same object Color.Black.
When,  "black", "silver", "white", ...,  rgb colors defined ... returns a new object.
I haven't a example list in mind but at least Color1 == Color2 is different according implementation.

Reply:
I think Phil is talking about just `return`ing the result, which doesn't change behavior. `""` can still `return Color.BLACK;` with the others creating a new object, ditching the oddly-placed temporary `Color` object.

Reply:
@SWinxy,
- if you're suggesting this
  ```
  } else {
              return colorNamed.get(strlc);
  ```
  We change the behavior (your comment) and lose the last test (color hex coded but without # prefix).
- or if you have in mind 
```
        } else {
                return new Color(colorNamed.get(strlc).color.getRGB(), true);
```
We preserve the behavior but still lose the last test (color hex coded but without # prefix).
- The way I interpreted @prrace comment :
  ```
          } else {
              Color color = colorNamed.get(strlc);
              if (color != null) {
                  return color;
  ```
  We change the behavior :
  - In the code before this PR. The results were : null, Color.Black if "", and new Color Object for all other colors (the 10 named colors and all hex coded colors).
  - In the first implementation achieved in this PR, like the code above, results changed to : null, Color.Black if "", the same Object for each (149) named-color or hex coded color.
  - After your comment, results became : null, Color.Black if "", and new Color Object for all other Color (149 named or hex coded).

Reply:
Mmm I completely forgot that we're using the map here, and for that we'll need to create a `new Color` from it. My b. There could be a check to `colorNames.has(str)` and return the `new Color(colorNames.get(str))` since there will never be a `null` returned for a valid key.